### ISSUE TITLE
Detective Toy Car can't be entered

### ISSUE THREAD

**Issue by CivilWarBuff** [TTT-2/TTT2#1627]
## Your version of TTT2 (mandatory)

- TTT2 v0.14.0b steam workshop

## Video

https://github.com/user-attachments/assets/0bcb09b6-9cdd-4321-9f18-6bca08bd04a5

## Describe the bug (mandatory)

- When attempting to enter the toy car it seems to immediately exit you from the vehicle

## To reproduce

Steps to reproduce the behaviour:

1. Purchase Detective Toy Car
2. Place it in the world
3. Attempting to enter by pressing "e" will not have the intended effect

## Expected behaviour

The expected behavior is to allow the player to enter the vehicle and drive it around

## Context (please provide as much as you can)

- Collection link of your Server: https://steamcommunity.com/sharedfiles/filedetails/?id=2913451164

**Comment by TimGoll** [comment]
I don't understand why it isn't working. I just spent two hours debugging it and it doesn't make any sense. You are sure it worked in the previous TTT2 version, arent' you? Because I'm unable to find any issue in our code. The `ent:Use` function is properly called for the correct entity. But then your linked mod has the issue that it seems to interpret it as leaving the vehicle and therefore teleports you.

Right now I'm at my wits end, not sure how I can fix this.

**Comment by CivilWarBuff** [comment]
Yeah that’s very strange and it definitely was working in the last version.
I tested it with my group shortly before the v0.14 release and had a lot of
fun with it. If it’s not an issue on TTT’s end I’ll try reaching out to the
mod developer and see if I get a response from them. Thanks for trying :)

**Comment by TimGoll** [comment]
I think it has something to do with the way how our use interaction is working. However this is the only entity where it breaks. And even though everything seems to work as intended, it just doesn't work

**Comment by Histalek** [comment]
Short status update on this:

We now know where we have broken this but not exactly why/how.

Additionally this seems to happen to `prop_vehicle_jeep` vehicles, but not all of them which is why we haven't caught this sooner. (`vehicle_prop_prisoner_pod` doesn't seem to be affected)

So we definitely still need to find out why this happens before we can fix this

**Comment by TimGoll** [comment]
Good news, Recent changes by @wgetJane in the linked PR "accidentally" also fixed this issue! And the issue with vehicles in general
